tests/ui/const-generics/issues/issue-68366.full.stderr

```diff
@@ -6,6 +6,11 @@ LL | impl <const N: usize> Collatz<{Some(N)}> {}
    |
    = note: expressions using a const parameter must map each value to a distinct output value
    = note: proving the result of expressions other than the parameter are unique is not supported
+help: Either remove the type parameter 'N' or make use of it, for example ` for S<N>`.
+   |
+LL - impl <const N: usize> Collatz<{Some(N)}> {}
+LL + impl <> Collatz<{Some(N)}> {}
+   |
```

Comment:
in this case the suggestion is wrong, removing it is never the right thing if it is used, but in a way that isn't legal.

Reply:
since we know it in this case (we are emitting the two notes above after all), you can re-use the condition for that and not emit the suggestion in that case